apps/web/components/posts/table.tsx

```diff
@@ -33,26 +48,37 @@ const Table = ({ posts }: TableProps) => {
                   </th>
                 </tr>
               </thead>
-              <tbody className="divide-y divide-gray-200 bg-white">
-                {posts.map((post) => (
+              
+              <tbody className="divide-y divide-gray-200 bg-white" > 
+                {posts.map((post) => ( //THE ONCLICK FOR FINDING THE POST ID + CONTENT has been moved so that the user can click delete button without going to other page
                   <tr
                     key={post.id.toString()}
-                    onClick={() => {
-                      push({ pathname: "post/[id]", query: { id: "HELLO" } });
-                    }}
                     className="cursor-pointer"
                   >
-                    <td className="whitespace-nowrap py-4 pl-4 pr-3 text-sm font-medium text-gray-900 sm:pl-6">
+                    <td className="whitespace-nowrap py-4 pl-4 pr-3 text-sm font-medium text-gray-900 sm:pl-6"
+                      onClick={() => {
+                        push({ pathname: "post/[id]", query: {content: post.content.toString(),  id: post.id.toString() } }); //PATH NAME + ID, CHANGED
```

Comment:
Here we can place the `onClick` in the `<tr>` row element of the table so we can go to the post by clicking anywhere in the row (shouldn't have two duplicate ones).
We also shouldn't put the content of the post into the query
```suggestion
                        push({ pathname: "post/[id]", query: { id: post.id.toString() } }); //PATH NAME + ID, CHANGED
```
This makes the URL confusing e.g. it looks like `http://localhost:3000/post/15?content=Lorem+ipsum+dolor+sit+amet%2C+consectetur+adipiscing+elit.+Quisque+blandit+sem+et+nulla+pharetra%2C+ut+lacinia+urna+aliquam.+Aliquam+fringilla+eros+ut+risus+ullamcorper+elementum+vitae+at+purus.`

What we want to do here is just put the `id` in the query, then in the actual `<Post>` component in `[id].tsx`, we make a call to the graphql api with the id to retrieve the contents of the post:
```jsx
const { query, push } = useRouter();
  const postId = String(query.id);

  const { data, loading } = useQuery<GetPostByIdResult, GetPostByIdVariables>(GET_POST_BY_ID, {
    variables: {
      getPostId: postId
    }
  });
```